source/opt/def_use_manager.cpp

```diff
@@ -76,33 +78,87 @@ const ir::Instruction* DefUseManager::GetDef(uint32_t id) const {
   return iter->second;
 }
 
-UseList* DefUseManager::GetUses(uint32_t id) {
-  auto iter = id_to_uses_.find(id);
-  if (iter == id_to_uses_.end()) return nullptr;
-  return &iter->second;
+DefUseManager::IdToUsersMap::const_iterator DefUseManager::UsersBegin(
+    const ir::Instruction* def) const {
+  return id_to_users_.lower_bound(
+      UserEntry(const_cast<ir::Instruction*>(def), nullptr));
 }
 
-const UseList* DefUseManager::GetUses(uint32_t id) const {
-  const auto iter = id_to_uses_.find(id);
-  if (iter == id_to_uses_.end()) return nullptr;
-  return &iter->second;
+bool DefUseManager::UsersNotEnd(const IdToUsersMap::const_iterator& iter,
+                                const IdToUsersMap::const_iterator& cached_end,
+                                const ir::Instruction* inst) const {
+  return (iter != cached_end && iter->first == inst);
+}
+
+bool DefUseManager::UsersNotEnd(const IdToUsersMap::const_iterator& iter,
+                                const ir::Instruction* inst) const {
+  return UsersNotEnd(iter, id_to_users_.end(), inst);
+}
+
+void DefUseManager::ForEachUser(const ir::Instruction* def,
+                                const std::function<void(ir::Instruction*)>& f) const {
+  // Ensure that |def| has been registered.
+  assert(def && def == GetDef(def->result_id()) && "Definition is not registered.");
+  auto end = id_to_users_.end();
+  for (auto iter = UsersBegin(def); UsersNotEnd(iter, end, def); ++iter) {
```

Comment:
line break after ';'

Reply:
I'm confused. Where do you think the line break is missing?